marlowe-playground-client/src/Marlowe/Gen.purs

```diff
@@ -33,7 +33,13 @@ genRational :: forall m. MonadGen m => MonadRec m => m Rational
 genRational = do
   n <- genBigInteger
   d <- genBigInteger
-  pure $ Rational n d
+  pure
+    -- this little fix is in Marlowe.Blockly so we need it here to make the tests work
+    
+    $ if d > zero then
+        Rational n d
+      else
+        Rational (-n) (-d)
```

Comment:
Hm, we have an instance of (Eq Rational) that deals with this. Then again, I am guessing some property failed that forced you to add this. It would be best to not have it though, cause users can write any Rational, so there are things we won't be testing.

Does a property fail without this? If so, could you tell me which?

Reply:
This is because of https://github.com/input-output-hk/plutus/blame/master/marlowe-playground-client/src/Marlowe/Blockly.purs#L1414 which makes code->blockly->code non-idempotent, this came up in the tests.

Reply:
I know, but I mean, I think the tests passed when I wrote this, I fixed it by adding an Eq instance that treats both Rationals the same, so I wonder why it failed now

Reply:
https://github.com/input-output-hk/plutus/blame/3d3b7ac8640204e0919740b5a41b87b86a2b86e3/marlowe-playground-client/src/Marlowe/Semantics.purs#L231

so they should be equal

Reply:
I can't remember why but before this PR it wasn't actually being tested. If you try it out on `master` you will see that `Scale` doesn't work in the playground. Not just this part but the parsing too. This PR makes sure a lot more things are tested and has thrown up quite a few bugs.

Reply:
Ah, its not needed now that I got the test to do an extra parse round-trip. Deleting.

Reply:
Sigh, after testing a lot a fiddling around I tracked down the issue. We need to do this because in tests where we wrap a `Rational` in a `TermWrapper` when we have two negative values then the column position of the term will different to if we have two positive values, even though the rationals themselves are equal